### DISCUSSION TITLE
Image Optimization still coming through even though I no longer use next/image

### DISCUSSION THREAD

**Discussion by JamesSingleton** [vercel/community#4200] [Help]
### Summary

I use Sanity CMS and since they can handle image optimization, I switched over to using the [sanity-image](https://www.npmjs.com/package/sanity-image) library as my website was hitting the 1,000 images almost every single time the period refreshed. I switched to this around September 5th and if you take a look at the image below, you will see that for Redshirt Sports, there are still requests coming through for it.

<img width="955" alt="Screenshot 2023-09-21 at 11 47 49 PM" src="https://github.com/vercel/community/assets/21000200/7da0fcba-e432-498d-9511-ef6da1b71308">

The only place I use `next/image` in my application is for `react-tweet` but all of these requests are for Sanity.io URLs.

### Steps to Reproduce

N/A

**Comment by amyegan** [comment]
Hi @JamesSingleton! Do you have a [minimal reproducible example](https://vercel.com/support/articles/creating-a-minimal-reproducible-example) that we can use to help with debugging?

**Reply by JamesSingleton** [reply]
@amyegan, I don't even know how I would provide a minimal reproducible example for this to be honest. The only place I am using `next/image` in my application is for `react-tweet` per the [documentation](https://react-tweet.vercel.app/next). Like I mentioned I moved away from using `next/image` on September 5th but somehow almost 50% of my image utilization is coming from this particular website.
![Screenshot 2023-10-17 at 9 08 13 PM](https://github.com/vercel/community/assets/21000200/46b6b615-f14b-4ed7-8093-a3f67e1cb44c)

For example, these images were optimized today:
* https://cdn.sanity.io/images/8pbt9f8w/production/22cabf9908a0fb6b176211e6ee6f71b1671de231-1920x1280.webp?rect=0,78,1920,1125&w=1024&h=600&fit=min&auto=format
* https://cdn.sanity.io/images/8pbt9f8w/production/888db9ff532ad9cb9cb2d1dd8e3483020b94c1cf-1920x1280.webp?rect=153,0,1707,1280&w=640&h=480&q=40&fit=min&auto=format
* https://cdn.sanity.io/images/8pbt9f8w/production/e59c1a002fc7761c394913a6686fe28f02f55c1b-1395x785.webp?rect=0,44,1395,698&w=400&h=200&q=40&fit=min&auto=format
* https://cdn.sanity.io/images/8pbt9f8w/production/fc77136e00b1a78b93ca6fdf916036cc5a19ec15-1920x1080.webp?rect=240,0,1440,1080&w=640&h=480&q=40&fit=min&auto=format

I don't know if there is some super old cache somewhere that is still referencing code that is using `next/image` or what.

**Reply by JamesSingleton** [reply]
@amyegan following up on this on what's the best approach to get some help on this. I am at 50% usage 
<img width="973" alt="Screenshot 2023-11-07 at 8 13 47 AM" src="https://github.com/vercel/community/assets/21000200/0b8f6061-d724-4245-b9ef-b7b2f73d4049">


I am still only using `next/image` in my `react-tweet` component like that documentation says. However, it's optimizing things like https://cdn.sanity.io/images/8pbt9f8w/production/da1457e3c70ff469e01d8bdb601827fb0860a886-1920x1080.webp?rect=0,60,1920,960&w=400&h=200&q=40&fit=min&auto=format

https://cdn.sanity.io/images/8pbt9f8w/production/ccff2cea01ff119554c769578f3f44035a57a2fd-2000x1333.jpg?rect=112,0,1777,1333&w=640&h=480&q=40&fit=min&auto=format

These articles were published over a year ago.

**Reply by amyegan** [reply]
I'm still not able to reproduce the issue in my test project. If you're able to provide a [minimal reproducible example](https://vercel.com/support/articles/creating-a-minimal-reproducible-example), that would help me investigate the issue. 

Without knowing more, the things I can recommend to try are:

- Add [`unoptimized`](https://nextjs.org/docs/app/api-reference/components/image#unoptimized) to the images
- Check if anything is accessing an older deployment's unique [generated URL](https://vercel.com/docs/deployments/generated-urls#)

**Reply by JamesSingleton** [reply]
Ah! How can I see if something is accessing an older deployment URL? That might be it, I remember seeing a generated URL showing up in some of my SEO reports before.

**Reply by JamesSingleton** [reply]
Hi @amyegan,

Circling back to this as it has been about 2 weeks and I am still seeing image optimization requests for a project that no longer uses `next/image` and was curious if there is any way to see what URL users are hitting in order to cause this image optimization request.

<img width="443" alt="Screenshot 2023-11-19 at 4 22 18 PM" src="https://github.com/vercel/community/assets/21000200/f56e6ca2-c42c-4427-88e0-0ccd03d63d94">

**Comment by JamesSingleton** [comment]
Hoping I can still get some help from Vercel support about image optimization requests coming in for a website that no longer uses Vercel for image optimization

<img width="325" alt="Screenshot 2024-02-20 at 8 10 48 AM" src="https://github.com/vercel/community/assets/21000200/7aec0836-9331-4f4c-be59-10b177cf5be7">
<img width="321" alt="Screenshot 2024-02-20 at 8 12 05 AM" src="https://github.com/vercel/community/assets/21000200/dbf3d85e-ab0f-44f5-8f23-675f8de4a01d">

**Reply by leerob** [reply]
You mention you're still using `next/image` for `react-tweet` - what does the list of source image show that are being optimized?

**Reply by leerob** [reply]
It seems they're still from Sanity - are there any preview deployments that are still receiving traffic? If you want to ensure this usage is removed entirely, then you would want to clean up any existing deployments that are still using Vercel Image Optimization uses what you've moved to.

**Reply by JamesSingleton** [reply]
Hi @leerob,

That is correct, I am only using it for `react-tweet` but the amount of articles that have a tweet in them are few and far between and one of them is listed as `unoptimized`

```tsx
const TweetComponents: TwitterComponents = {
  AvatarImg: (props) => <Image {...props} alt={props.alt} />,
  MediaImg: (props) => <Image {...props} alt={props.alt} fill unoptimized />,
}
```

The ones that are showing up in Vercel are sanity links. For example, from yesterday there was one image that is from an article posted on 1/25/2022 and according to Plausible, has not gotten any traffic in the last 7 days. Regardding any preview deployments, how do I check that? I no longer have the `vercel.app` URL under my domains as I deleted that a couple of weeks ago by now. Is it just anything that is under "Active Branches"?


<img width="1344" alt="Screenshot 2024-02-20 at 8 43 38 AM" src="https://github.com/vercel/community/assets/21000200/874d5feb-18b7-4a89-aff7-678416ea7310">

**Reply by JamesSingleton** [reply]
@leerob, just circling back to this as I am nearing 63% image utilization with a majority of them coming from Redshirt Sports but ideally shouldn't be.

**Reply by leerob** [reply]
Are the source images shown on the usage page coming from Sanity as mentioned above? You can filter down to the Redshirt Sports project, but there is not the ability on the usage page to filter to a specific deployment. What that is telling you is that the usage is coming from a deployment on the Redshirt Sports project.

**Reply by JamesSingleton** [reply]
That ones for Redshirt Sports are coming from Sanity 

```
https://cdn.sanity.io/images/${my_project_id}/production/43cba752054c8d84b3b13662717eff242a32ac44-1536x2048.webp?rect=0,256,1536,1536&w=72&h=72&q=60&fit=min&auto=format
```

Once a close a PR does that automatically delete that deployment URL? Also, how are people getting these deployment URLs? My project is private. The only PR I had open I just closed, but even that one wasn't using the `next/image` component. 

So does this mean that somewhere someone is accessing a super old deployment URL that was never cleaned up?

**Reply by leerob** [reply]
Closing a PR does not delete the Preview Deployment. Are you confident there is no usage of `next/image` that is optimizing images in your project? If you want to narrow it down further, you can delete past preview deployments from the dashboard or through the API.

**Reply by JamesSingleton** [reply]
These are the only references for `next/image` in my repo

<img width="641" alt="Screenshot 2024-02-28 at 6 22 27 PM" src="https://github.com/vercel/community/assets/21000200/903fa348-37bb-489a-b4eb-dc8124ac13f5">

with that second one being

```tsx
const TweetComponents: TwitterComponents = {
  AvatarImg: (props) => <Image {...props} alt={props.alt} />,
  MediaImg: (props) => <Image {...props} alt={props.alt} fill unoptimized />,
}
```

And these are not Tweet images, they are Sanity images. Oh I am going to have to manually click into every single deployment preview and delete them 🙈 what is the API route? Preview deployments go back 853 days for me.

**Reply by JamesSingleton** [reply]
I deleted the preview deployments from what was showing as "Active" for previous PRs that I had opened but closed 🤞🏼 that was the cause but should probably clean up 800+ days of preview URLs

**Reply by JamesSingleton** [reply]
So even though I deleted the ones listed as "active" I am still getting optimizations for this particular website 4 days afterwards. I am starting to think 1 of 2 things is happening:

1. There is some super old cached version floating around somewhere that keeps getting hit
2. One of my preview URLs that wasn't under "active" is getting hit. Unfortunately, I don't know if there is an _easy_ way to figure that out.

**Reply by JamesSingleton** [reply]
Hi @leerob,

Just circling back to this. I was hopeful that maybe deleting the "Active Branches" would have worked. Unfortunately, it looks like I keep climbing closer and closer to that hard limit. I would prefer not to have to manually go through and delete the probably 300+ preview URLs 😓 

<img width="959" alt="Screenshot 2024-03-06 at 8 41 58 AM" src="https://github.com/vercel/community/assets/21000200/7b44ed11-a62d-49be-a028-2461014a7a30">

<img width="323" alt="Screenshot 2024-03-06 at 8 40 19 AM" src="https://github.com/vercel/community/assets/21000200/3d64a7fd-5766-4d55-94c0-77ad37d654a9">
<img width="319" alt="Screenshot 2024-03-06 at 8 40 43 AM" src="https://github.com/vercel/community/assets/21000200/b52b1341-b92d-4073-bf7c-2c2815ebf30f">
<img width="1283" alt="Screenshot 2024-03-06 at 8 41 27 AM" src="https://github.com/vercel/community/assets/21000200/726eff9a-0a77-4b0f-8b51-12d20221a9dc">

**Comment by JamesSingleton** [comment]
I am going to go ahead and close this issue out. Just an FYI @leerob and @amyegan, after talking with Malte I turned on deployment protection around May 9th. Since then the image optimizations coming from this particular project have dropped significantly. It would be nice to see WHICH deployment branches were getting outside traffic though so I could go through and delete them.

![Screenshot 2024-05-19 at 9 02 13 AM](https://github.com/vercel/community/assets/21000200/957b65e8-c1b3-44eb-a8da-b931e3a1e534)
